src/main/java/seedu/exercise/model/Model.java

```diff
@@ -147,4 +147,20 @@
      * @throws NullPointerException if {@code predicate} is null.
      */
     void updateFilteredRegimeList(Predicate<Regime> predicate);
+
+    /**
+     * Returns an unmodifiable view of the list of suggested exercises
+     * */
+    ObservableList<Exercise> getSuggestedExerciseList();
+
+    /**
+     * Updates the filter of the filtered exercise list to filter by the give {@code predicate}.
+     * @throws NullPointerException if {@code predicate} is null.
+     */
+    void updateSuggestedExerciseList(Predicate<Exercise> predicateShowAllExercises);
+
+    /**
+     * Returns the data of all exercises in the database
+     */
+    ReadOnlyExerciseBook getAllExerciseBookData();
```

Comment:
To differentiate from the normal `getUserExerciseBookData`, maybe give the DB methods a name like `getDatabaseData` or `getExerciseDatabase`

Reply:
Ok